File: myspark2/src/main/java/com/oldboy/java/sparksql/MySparkSql.java
package com.oldboy.java.sparksql;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class MySparkSql {
    public static void main(String[] args) {
        System.setProperty("user.name", "centos");
        System.setProperty("HADOOP_USER_NAME", "centos");


        SparkSession sess = SparkSession.builder()
                //.config("spark.yarn.jars", "hdfs://mycluster/spark/jars/*")
                .master("spark://s101:7077")
                .appName("app")
                .enableHiveSupport()
                .getOrCreate();

        //Dataset<Row> sql = sess.sql("select count(distinct(cookie)) , month,day, grouping_id()  from big14.uv2 group by month,day with cube order by grouping_id()");

        sess.sql("use big14");
        Dataset<Row> sql = sess.sql("show tables");

        sql.show();

    }
}

File: myspark/src/main/java/com/oldboy/scala/sparksql/MySparkSql.scala
package com.oldboy.scala.sparksql

import java.util

import org.apache.spark.SparkConf
import org.apache.spark.sql.types._
import org.apache.spark.sql.{Row, SparkSession}

object MySparkSql extends Serializable {

    def main(args: Array[String]): Unit = {
        System.setProperty("user.name", "centos");
        System.setProperty("HADOOP_USER_NAME", "centos");

        val conf = new SparkConf()
        conf.set("spark.yarn.jars", "hdfs://mycluster/spark/jars/*")
        conf.setMaster("spark://s101:7077")
            .setAppName("app")

        val sess = SparkSession.builder()
            .config(conf)
            .enableHiveSupport()
            .getOrCreate()

        import scala.collection.JavaConversions._
        import sess.implicits._
        val list: util.List[Row] = List(
            Row(1, "tom", 20),
            Row(2, "tomas", 40),
            Row(3, "tomson", 60),
            Row(4, "tomason", 80),
            Row(5, "tomasLee", 90)
        )

        val schema = StructType(List[StructField](
            StructField("id", IntegerType, false),
            StructField("name", StringType, true),
            StructField("age", IntegerType, true))
        )
        val df1 = sess.createDataFrame(list, schema)
        df1.createOrReplaceTempView("_user")
        //df1.show()


        val df2 = sess
            .read
            .option("header", "false") //处理csv文件首行问题
            .option("delimiter", "\t") //处理分隔符问题，默认是 ","
            .option("charset", "UTF-8")
            .csv("/orders.txt")
            .toDF("oid", "orderno", "oprice", "uid")

        df2.createTempView("_order")

        val df = sess.sql("select a.id, a.name, a.age,b.orderno,b.oprice from `_user` a left join `_order` b on a.id = b.uid")
        df.show()
    }
}
